### PR TITLE
allow syncing from project config

### PR DESCRIPTION
**Pull request by rcannood**
## New functionality

* `sync-and-cache`: Added an action for syncing and caching test resources (PR #31).

## Bug fixes

* `check-concurrent-pr`: Added input parameter `github_token` for checking concurrent PRs (PR #31).

### CODE REVIEW

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: project/sync-and-cache/action.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          while read -r line; do
+            type=$(echo "$line" | yq e '.type')
+            path=$(echo "$line" | yq e '.path')
+            dest=$(echo "$line" | yq e '.dest')
+            if [ "$type" == "s3" ]; then
+              hash=$(hash_s3 "$path")
+            fi
+            echo "dest: $dest, hash: $hash"
+            hashes+=( "dest: $dest, hash: $hash" )
+          done
```
[the remark this answers is not in the document]
Comment (rcannood) [reply project/sync-and-cache/action.yml:81]: Both work:

```bash
variable="foo
bar
zing
dong
dang"

echo "# ATTEMPT 1:"
echo "$variable" | \
  while read line; do
    echo "line1: $line"
  done
```

    # ATTEMPT 1:
    line1: foo
    line1: bar
    line1: zing
    line1: dong
    line1: dang

```bash
echo "# ATTEMPT 2:"
while read line; do
  echo "line2: $line"
done <<< "$variable"
```

    # ATTEMPT 2:
    line2: foo
    line2: bar
    line2: zing
    line2: dong
    line2: dang

though I think I prefer the first option, especially if there are more pipes involved (`echo ... | while ... done | md5sum | awk ...`)

File: project/sync-and-cache/action.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+runs:
+  using: 'composite'
+  steps:
+    - name: Test resources
+      id: test_resources
+      shell: bash
+      run: |
+        # output multiline string
+        echo "test_resources<<EOF" >> $GITHUB_OUTPUT
+        yq e '.info.test_resources[] | "{type: " + (.type // "s3") + ", path: " + .path + ", dest: " + .dest + "}"' "${{ inputs.project_config }}" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (rcannood) [reply project/sync-and-cache/action.yml]: Added info to the docs!

File: project/sync-and-cache/action.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        echo "EOF" >> $GITHUB_OUTPUT
+
+        echo "dest_paths<<EOF" >> $GITHUB_OUTPUT
+        yq e '.info.test_resources[] | .dest' "${{ inputs.project_config }}" >> $GITHUB_OUTPUT
+        echo "EOF" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (rcannood) [reply project/sync-and-cache/action.yml]: Fixed!

**Review by DriesSchaumont — APPROVED** [review]
LGTM!

**Pull request merged**